extensions/wikia/Email/Controller/EmailConfirmationController.class.php

```diff
@@ -44,151 +50,63 @@ public function body() {
+		return $this->getMessage( 'emailext-emailconfirmation-footer-1' )->text();
 	}
 
-	protected function getEmailSpecificFooterMessages() {
+	private function createContentFooterList() {
 		return [
-			$this->getMessage( 'emailext-emailconfirmation-footer-1' )->text(),
-			$this->getMessage( 'emailext-emailconfirmation-footer-2' )->text()
+			$this->getMessage( 'emailext-emailconfirmation-footer-list-1' )->text(),
+			$this->getMessage( 'emailext-emailconfirmation-footer-list-2' )->text(),
+			$this->getMessage( 'emailext-emailconfirmation-footer-list-3' )->text(),
+			$this->getMessage( 'emailext-emailconfirmation-footer-list-4' )->text(),
 		];
-
-	}
-}
-
-class EmailConfirmationReminderController extends AbstractEmailConfirmationController {
-
-	protected function getSubject() {
-		return $this->getMessage( 'emailext-emailconfirmation-reminder-subject', $this->getTargetUserName() )->parse();
 	}
 
-	protected function getSummary() {
-		return $this->getMessage( 'emailext-emailconfirmation-reminder-summary' )->text();
+	private function getContentFooterMessagesBottom() {
+		$commonFooterMessage = $this->getCommonFooterMessages();
+		$emailSpecificFooterMessages = $this->getEmailSpecificFooterMessages();
+		return array_merge( $emailSpecificFooterMessages, $commonFooterMessage );
 	}
 
-	protected function getEmailSpecificFooterMessages() {
+	private function getCommonFooterMessages() {
 		return [
-			$this->getMessage( 'emailext-emailconfirmation-reminder-footer-1',
-				$this->getTargetUserName() )->parse()
+			$this->getMessage( 'emailext-emailconfirmation-community-team' )->text()
 		];
 	}
-}
-
-class ConfirmationChangedEmailController extends AbstractEmailConfirmationController {
-
-	private $newEmail;
-
-	public function initEmail() {
-		parent::initEmail();
-
-		$this->newEmail = $this->request->getVal( 'newEmail' );
-		$this->assertValidChangedParams();
-	}
-
-	protected function assertValidChangedParams() {
-		if ( empty( $this->newEmail ) ) {
-			throw new Check( "A value must be passed for parameter 'newEmail'" );
-		}
-	}
-
-	protected function getTargetUserEmail() {
-		return $this->newEmail;
-	}
-
-	protected function getSubject() {
-		return $this->getMessage( 'emailext-emailconfirmation-changed-subject' )->text();
-	}
-
-	protected  function getSummary() {
-		return $this->getMessage( 'emailext-emailconfirmation-changed-summary' )->text();
-	}
 
 	protected function getEmailSpecificFooterMessages() {
 		return [
-			$this->getMessage( 'emailext-emailconfirmation-changed-footer-1' )->text(),
-			$this->getMessage( 'emailext-emailconfirmation-changed-footer-2' )->text(),
+			$this->getMessage( 'emailext-emailconfirmation-footer-2' )->text()
 		];
 	}
 
 	protected static function getEmailSpecificFormFields() {
-		$parentForm = parent::getEmailSpecificFormFields();
-
-		$parentForm['inputs'][] = [
-			'type' => 'text',
-			'name' => 'newEmail',
-			'label' => "New Email",
-			'tooltip' => "The user's new email",
+		$form = [
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$form". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)